frontend/src/utils/safeHref.test.ts

```
@@ -0,0 +1,55 @@
+  it('refuses a value that is not a url at all, and nothing', () => {
+    expect(safeHref('not a url')).toBeNull();
+    expect(safeHref('')).toBeNull();
+    expect(safeHref(null)).toBeNull();
+    expect(safeHref(undefined)).toBeNull();
+  });
+
+  it('keeps an absolute http(s) url, in the form the parser read', () => {
+    expect(safeHref(WIKIVOYAGE_PAGE)).toBe(WIKIVOYAGE_PAGE);
+    // `http` as well as `https`: a licence url from old wiki metadata is one,
+    // and neither scheme executes.
+    expect(safeHref('HTTP://creativecommons.org/licenses/by-sa/4.0/'))
```

ESLint

[error] Using http protocol is insecure. Use https instead.

(sonarjs/no-clear-text-protocols)